src/SocketTrebuchet.ts

```diff
@@ -45,6 +55,39 @@ class SocketTrebuchet extends Trebuchet {
     }, this.timeout * 1.5)
   }
 
+  private sendAck (mid: number) {
+    const ack = new Uint8Array(4)
+    const view = new DataView(ack.buffer)
+    // first bit is a 0 if ACK
+    const ackId = mid << 1
+    // guarantee little endian
+    view.setUint32(0, ackId, true)
+    this.ws.send(ack)
+  }
+
+  private sendReq (mid: number) {
+    const req = new Uint8Array(4)
+    const view = new DataView(req.buffer)
+    // first bit is 0 if REQ
+    const reqId = (mid << 1) | 1
```

Comment:
its the client's responsibility to ask for what it doesn't have. that way the server doesn't have to keep track of a ton of timers.

note that if we do away with the timeouts on the server, the client doesn't know it's missing message `n` until message `n + 1` comes in. i think that's fine for now